### ISSUE TITLE
Error after upgrading to 8.0.2: TypeError: Cannot read property 'match' of undefined

### ISSUE THREAD

**Issue by kaiyoma** [babel/babel-eslint#541]
eslint version: 4.10.0
Node version: 8.9.0
yarn version: 1.3.2

I'm currently running babel-eslint 8.0.1 and everything works great. If I upgrade to 8.0.2 and run my linting script, I get this error:

```
Cannot read property 'match' of undefined
TypeError: Cannot read property 'match' of undefined
    at Literal (...\node_modules\eslint\lib\rules\no-octal-escape.js:32:40)
    at listeners.(anonymous function).forEach.listener (...\node_modules\eslint\lib\util\safe-emitter.js:47:58)
    at Array.forEach (<anonymous>)
    at Object.emit (...\node_modules\eslint\lib\util\safe-emitter.js:47:38)
    at NodeEventGenerator.applySelector (...\node_modules\eslint\lib\util\node-event-generator.js:251:26)
    at NodeEventGenerator.applySelectors (...\node_modules\eslint\lib\util\node-event-generator.js:280:22)
    at NodeEventGenerator.enterNode (...\node_modules\eslint\lib\util\node-event-generator.js:294:14)
    at CodePathAnalyzer.enterNode (...\node_modules\eslint\lib\code-path-analysis\code-path-analyzer.js:606:23)
    at Traverser.enter (...\node_modules\eslint\lib\linter.js:955:32)
    at Traverser.__execute (...\node_modules\estraverse\estraverse.js:397:31)
```

I don't know what code is triggering this because of the error. I'm not aware of any octal escape sequences in the codebase.

**Comment by KagamiChan** [comment]
quoted from eslint:
> As our AST spec, Literal node should have raw property.

Please fix this

**Comment by nicolo-ribaudo** [comment]
Can you provide a code example which triggers this error?

**Comment by kaiyoma** [comment]
The plugin is failing on lots of files in our codebase, but here's one example:

```
import PropTypes from 'prop-types';
import React from 'react';
import PieChart from 'react-minimal-pie-chart';

const AnnotatedPieChart = props => {
  return (
    <div className='pie-chart'>
      {props.total ?
        <PieChart
          data={props.data}
          lineWidth={30}
          paddingAngle={1}
        /> :
        <PieChart
          data={[{ value: 1 }]}
          lineWidth={30}
        />
      }
      <div className='pie-chart-total'>
        {props.total}
      </div>
    </div>
  );
};

AnnotatedPieChart.propTypes = {
  data: PropTypes.array,
  total: PropTypes.number,
};

export default AnnotatedPieChart;
```

**Comment by kaiyoma** [comment]
Funny, this issue seemed to go away for a while after upgrading, but now it's back. We've been running version 8.2.1 for a little while and everything was great, but as soon as I upgrade to 8.2.2, I get a stack trace when running our tests:

```
Cannot read property 'match' of undefined
TypeError: Cannot read property 'match' of undefined
    at Literal (...\node_modules\eslint\lib\rules\no-octal-escape.js:33:40)
    at listeners.(anonymous function).forEach.listener (...\node_modules\eslint\lib\util\safe-emitter.js:47:58)
    at Array.forEach (<anonymous>)
    at Object.emit (...\node_modules\eslint\lib\util\safe-emitter.js:47:38)
    at NodeEventGenerator.applySelector (...\node_modules\eslint\lib\util\node-event-generator.js:251:26)
    at NodeEventGenerator.applySelectors (...\node_modules\eslint\lib\util\node-event-generator.js:280:22)
    at NodeEventGenerator.enterNode (...\node_modules\eslint\lib\util\node-event-generator.js:294:14)
    at CodePathAnalyzer.enterNode (...\node_modules\eslint\lib\code-path-analysis\code-path-analyzer.js:608:23)
    at Traverser.enter [as _enter] (...\node_modules\eslint\lib\linter.js:1017:32)
    at Traverser._traverse (...\node_modules\eslint\lib\util\traverser.js:132:14)
```

**Comment by dancon** [comment]
I hava the same issue with version 8.2.2.

```
Cannot read property 'match' of undefined
TypeError: Cannot read property 'match' of undefined
    at Literal (../node_modules/eslint/lib/rules/no-octal-escape.js:33:40)
    at listeners.(anonymous function).forEach.listener (../node_modules/eslint/lib/util/safe-emitter.js:47:58)
    at Array.forEach (<anonymous>)
    at Object.emit (../node_modules/eslint/lib/util/safe-emitter.js:47:38)
    at NodeEventGenerator.applySelector (../node_modules/eslint/lib/util/node-event-generator.js:251:26)
    at NodeEventGenerator.applySelectors (../node_modules/eslint/lib/util/node-event-generator.js:280:22)
    at NodeEventGenerator.enterNode (../node_modules/eslint/lib/util/node-event-generator.js:294:14)
    at CodePathAnalyzer.enterNode (../node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:608:23)
    at Traverser.enter [as _enter] (../node_modules/eslint/lib/linter.js:1017:32)
    at Traverser._traverse (../node_modules/eslint/lib/util/traverser.js:132:14)
```

**Comment by KagamiChan** [comment]
Could confirm this shows up again, please consider fix it

**Comment by leolord** [comment]
Same issue with 8.2.2. degrading to 8.2.1 solves the issue.

**Comment by justinyueh** [comment]
I have the same issue with v8.2.2, v8.2.3 solved this issue.

**Comment by kaiyoma** [comment]
Same here, things are working again with 8.2.3.
